src/proto/colors.rs

```diff
@@ -0,0 +1,45 @@
+//! An extension trait that provides the ability to strip IRC colors from a string
+use regex::Regex;
+
+/// An extension trait giving strings a function to strip IRC colors
+pub trait FormattedString {
+
+    /// Returns true if the string contains color, bold, underline or italics
+    fn is_formatted(&self) -> bool;
+
+    /// Returns the string with all color, bold, underline and italics stripped
+    fn strip_colors(&self) -> String;
```

Comment:
I think names like `strip_formatting` or `strip_irc_formatting` better reflects the effect on the string (given bold, underline, italics).